oshi-core/src/main/java/oshi/util/platform/windows/RegistryUtil.java

```diff
@@ -0,0 +1,181 @@
+                    || value instanceof byte[]) {
+                return value;
+            }
+        } catch (Win32Exception e) {
+            LOG.trace("Unable to access " + path + " with flag " + accessFlag + ": " + e.getMessage());
+        } finally {
+            if (hKey != null) {
+                int rc = ADV.RegCloseKey(hKey);
+                if (rc != ERROR_SUCCESS) {
+                    throw new Win32Exception(rc);
+                }
+            }
+        }
+        return null;
+    }
+
+    private static HKEY getRegistryHKey(HKEY rootKey, String path, int accessFlag) {
+        HKEYByReference phkKey = new HKEYByReference();
+        int rc = ADV.RegOpenKeyEx(rootKey, path, 0, KEY_READ | accessFlag, phkKey);
+        if (rc != ERROR_SUCCESS) {
+            throw new Win32Exception(rc);
+        }
+        return phkKey.getValue();
+    }
+
+    private static long registryValueToLong(Object val) {
+        if (val == null) {
+            return 0L;
+        }
+
+        // Calculate reasonable timestamp bounds (current time to 30 years ago)
+        long currentTimeSecs = System.currentTimeMillis() / 1000L;
+        long minSaneTimestamp = currentTimeSecs - THIRTY_YEARS_IN_SECS;
+        if (val instanceof Integer) {
+            int value = (Integer) val;
+            if (value > minSaneTimestamp && value < currentTimeSecs) {
+                return value * 1000L;
+            }
+            return value;
+        } else if (val instanceof String) {
+            String dateStr = ((String) val).trim();
+            // Try yyyyMMdd first
+            long epoch = ParseUtil.parseDateToEpoch(dateStr, "yyyyMMdd");
+            if (epoch == 0) {
+                // If that fails, try MM/dd/yyyy
+                epoch = ParseUtil.parseDateToEpoch(dateStr, "MM/dd/yyyy");
+            }
+            return epoch;
+        }
+        return 0L;
+    }
+
+    /**
+     * Returns a registry value as a String. (without access flag) Currently supports String and Binary
+     */
+    public static String getStringValue(WinReg.HKEY root, String path, String key) {
+        try {
+            Object val = Advapi32Util.registryGetValue(root, path, key);
+            return registryValueToString(val);
+        } catch (Win32Exception e) {
+            LOG.trace("Unable to access " + path + ": " + e.getMessage());
+            return null;
+        }
+    }
+
+    /**
+     * Returns a registry value as a String. (with access flag) Currently supports String and Binary
+     */
+    public static String getStringValue(WinReg.HKEY root, String path, String key, int accessFlag) {
+        Object val = getRegistryValueOrNull(root, path, key, accessFlag);
+        return registryValueToString(val);
+    }
+
+    /**
+     * Decodes registry value to String using multiple fallback encodings.
+     */
+    private static String registryValueToString(Object val) {
+        if (val == null)
+            return null;
```

Comment:
Style nit (and elsewhere):
```suggestion
        if (val == null) {
            return null;
        }
```